storagenode/contact/chore.go

```diff
@@ -48,53 +58,108 @@ func (chore *Chore) Run(ctx context.Context) (err error) {
 	defer mon.Task()(&ctx)(&err)
 	chore.log.Info("Storagenode contact chore starting up")
 
+	var group errgroup.Group
+
 	if !chore.service.initialized.Wait(ctx) {
 		return ctx.Err()
 	}
+	chore.mu.Lock()
+	for _, satellite := range chore.trust.GetSatellites(ctx) {
+		satellite := satellite
+		backOff := time.Second
+		interval := chore.interval
+		cycle := sync2.NewCycle(interval)
+		chore.cycles = append(chore.cycles, cycle)
+
+		cycle.Start(ctx, &group, func(ctx context.Context) error {
+			chore.log.Info("starting contact cycle for satellite " + satellite.String())
+			err := chore.pingSatellite(ctx, satellite)
+			if err == nil {
+				return nil
+			}
 
-	return chore.Loop.Run(ctx, func(ctx context.Context) error {
-		if err := chore.pingSatellites(ctx); err != nil {
-			chore.log.Error("pingSatellites failed", zap.Error(err))
-		}
-		return nil
-	})
+			interval := backOff
+			retries := 0
+			for {
+				if !sync2.Sleep(ctx, interval) {
+					chore.log.Error("ping satellite failed after " + strconv.Itoa(retries) + " retries timed out")
+					return ctx.Err()
```

Comment:
I think returning an error breaks the cycle of out its loop
https://github.com/storj/storj/blob/master/internal/sync2/cycle.go#L88